pkg/providers/snow/fetch.go

```diff
@@ -56,3 +61,40 @@ func oldWorkerMachineTemplate(ctx context.Context, kubeclient kubernetes.Client,
 	}
 	return mt, nil
 }
+
+func oldKubeadmConfigTemplate(ctx context.Context, kubeclient kubernetes.Client, clusterSpec *cluster.Spec, md *clusterv1.MachineDeployment) (*bootstrapv1.KubeadmConfigTemplate, error) {
+	if md == nil {
+		return nil, nil
+	}
+
+	kct := &bootstrapv1.KubeadmConfigTemplate{}
+	err := kubeclient.Get(ctx, md.Spec.Template.Spec.Bootstrap.ConfigRef.Name, constants.EksaSystemNamespace, kct)
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return kct, nil
+}
+
+func oldEksaCluster(ctx context.Context, kubeclient kubernetes.Client, clusterSpec *cluster.Spec) (*v1alpha1.Cluster, error) {
+	cluster := &v1alpha1.Cluster{}
+	err := kubeclient.Get(ctx, clusterSpec.Cluster.GetName(), constants.DefaultNamespace, cluster)
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return cluster, nil
+}
+
+func oldWorkerNodeGroupsMap(ctx context.Context, kubeClient kubernetes.Client, clusterSpec *cluster.Spec) (map[string]v1alpha1.WorkerNodeGroupConfiguration, error) {
+	eksaCluster, err := oldEksaCluster(ctx, kubeClient, clusterSpec)
```

Comment:
I'm a bit concerned about this
What's the purpose of checking the "old" cluster?

The concerns comes from the fact that in the controller, you won't be able to get the old version, since by the time we hit this code, it has already been updated 

We should try to no rely on the "past" state of any eksa object